### ISSUE TITLE
Unsubscribe of configuration API.

### ISSUE THREAD

**Issue by LaurenceLiZhixin** [dapr/dapr#3906]
I'm working on configuration API of multi-language sdks https://github.com/dapr/go-sdk/pull/200, and found subscription problems of configuration api https://github.com/dapr/dapr/pull/3903 . Now the implement of subscribe API is server-side-grpc-stream, and once the subscription is called, the return statement of server side causes the stream ends, and user's client can't get update event from config state store.
```proto
 rpc SubscribeConfigurationAlpha1(SubscribeConfigurationRequest) returns (stream SubscribeConfigurationResponse) {}
```
I would fix this bug, and encountered a new problem. After subscription function blocking, it would never be restart. How can user unsubscribe (stop) the blocking go routine if his app is shutdown, or the grpc connection broken? I think there is two way to solve, one is to add a new api called unsubscribe, the second is change the server-side-grpc-stream to a full duplex stream, and dapr runtime can know if the subscription should be shutdown, or the streaming is broken, and return the broking function.
@yaron2 @artursouza @daixiang0

**Comment by yaron2** [comment]
I would always encourage the explicit (and in this case, also non-breaking) API design. So, I support adding an Unsubscribe method.

**Comment by LaurenceLiZhixin** [comment]
OK, I'll follow up.

**Comment by greenie-msft** [comment]
Doc PR for this: https://github.com/dapr/docs/pull/2010

**Comment by artursouza** [comment]
Moving to 1.7 since PR is still in progress.

**Comment by LaurenceLiZhixin** [comment]
My colleague Lan Luo raised a problem with the existing design. My current design is based on the config's key granularity model. For example, if there are multiple subscribe operations of the configuration of key "myKey", one unsubscribe operation for "myKey" will impact all previous subscribe operations.
But with a granularity of the whole subscribe operation, each call to subscribe returns a specific ID that the user needs to pass when unsubscription, thus terminating only this subscribe without any other Subscribe Goroutine being affected.

My current scheme requires the user to guarantee that the key can be unsubscribed only if the user really does not need to subscribe to it. And the new idea that unsubscribe by ID is more flexible I think, and the user needs to record each subscription ID. 

Yaron, what do you think about? I will follow up . @yaron2

**Comment by LaurenceLiZhixin** [comment]
My colleague Lan Luo raised a problem with the existing design. My current design is based on the config's key granularity model. For example, if there are multiple subscribe operations of the configuration of key "myKey", one unsubscribe operation for "myKey" will impact all previous subscribe operations.
But with a granularity of the whole subscribe operation, each call to subscribe returns a specific ID that the user needs to pass when unsubscription, thus terminating only this subscribe without any other Subscribe Goroutine being affected.

My current scheme requires the user to guarantee that the key can be unsubscribed only if the user really does not need to subscribe to it. And the new idea that unsubscribe by ID is more flexible I think, and the user needs to record each subscription ID. 

Yaron, what do you think about? I will follow up . @yaron2

**Comment by yaron2** [comment]
@LaurenceLiZhixin returning a subscription ID is valid, and I agree that the API needs to support multiple subscriptions from the same app.

**Comment by yaron2** [comment]
Closed via #4130.
